plugin/hover.py

```diff
@@ -89,7 +89,7 @@ def diagnostics_content(self, diagnostics):
             formatted.append("</div>")
 
         formatted_warnings = list(
-            "<pre>{}</pre>".format(diagnostic.message)
+            "<pre>{}</pre>".format("[{}] {}".format(diagnostic.source, diagnostic.message) if diagnostic.source else "{}".format(diagnostic.message))
```

Comment:
The else case can be just `diagnostic.message` - or perhaps we want to only have maximum 1 format call per diagnostic?